site/docs/latest/advanced/Encryption.md

```diff
@@ -0,0 +1,157 @@
+---
+title: Pulsar Encryption
+lead: Encrypt messages published to pulsar
+tags:
+- crypto
+- encryption
+- clients
+---
+
+# Pulsar Encryption
+
+Pulsar encryption allows applications to encrypt messages at the producer and decrypt at the consumer. Encryption is performed using the public/private key pair configured by the application. Encrypted messages can only be decrypted by consumers with a valid key.
+
+## Asymmetric and symmetric encryption
+
+Pulsar uses dynamically generated symmetric AES key to encrypt messages(data). The AES key(data key) is encrypted using application provided ECDSA/RSA key pair, as a result there is no need to share the secret with everyone.
+
+Key is a public/private key pair used for encryption/decryption. The producer key is the public key, and the consumer key is the private key of the key pair.
+
+The application configures the producer with the public  key. This key is used to encrypt the AES data key. The encrypted data key is sent as part of message header. Only entities with the private key(in this case the consumer) will be able to decrypt the data key which is used to decrypt the message.
+
+A message can be encrypted with more than one key.  Any one of the keys used for encrypting the message is sufficient to decrypt the message
+
+Pulsar does not store the encryption key anywhere in the pulsar service. If you lose/delete the private key, your message is irretrievably lost, and is unrecoverable
+
+## Producer
+![alt text](https://github.com/apache/incubator-pulsar/blob/master/site/img/PulsarEncryptionProducer.jpg "Pulsar Encryption Producer")
+
+## Consumer
+![alt text](https://github.com/apache/incubator-pulsar/blob/master/site/img/PulsarEncryptionConsumer.jpg "Pulsar Encryption Consumer")
+
+## Here are the steps to get started:
+
+1. Create your ECDSA or RSA public/private key pair.
+
+    openssl ecparam -name secp521r1 -genkey -param_enc explicit -out test_ecdsa_privkey.pem
+    openssl ec -in test_ecdsa_privkey.pem -pubout -outform pkcs8 -out test_ecdsa_pubkey.pem
```

Comment:
I think these commands should be enclosed by ```.
```
1. Create your ECDSA or RSA public/private key pair.
    ```shell
    $ openssl ecparam -name secp521r1 -genkey -param_enc explicit -out test_ecdsa_privkey.pem
    $ openssl ec -in test_ecdsa_privkey.pem -pubout -outform pkcs8 -out test_ecdsa_pubkey.pem
    ```
2. Add the public and private key to the key management and configure your producers to retrieve public keys and consumers clients to retrieve private keys.
```